File: api/models/ConservationStatus.js
module.exports = (sequelize, DataTypes) => {
  const ConservationStatus = sequelize.define(
    "ConservationStatus",
    {
      id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
      },
      name: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      description: {
        type: DataTypes.TEXT,
      },
      severity: {
        type: DataTypes.STRING,
      },
    },
    {
      tableName: "ConservationStatus",
      timestamps: false,
    },
  );

  return ConservationStatus;
};

File: app/lib/models/ConservationStatus.dart
class ConservationStatus {
  final String description;
  final String severity;
  final String name;

  ConservationStatus({
    required this.description,
    required this.severity,
    required this.name,
  });

  factory ConservationStatus.fromJson(Map<String, dynamic> json) {
    return ConservationStatus(
      description: json['description'],
      severity: json['severity'],
      name: json['name'],
    );
  }
}
